addons/mail/static/src/models/autocomplete_input_view/autocomplete_input_view.js

```diff
@@ -0,0 +1,229 @@
+/** @odoo-module **/
+
+import { registerModel } from '@mail/model/model_core';
+import { attr, one } from '@mail/model/model_field';
+import { clear } from '@mail/model/model_field_command';
+
+registerModel({
+    name: 'AutocompleteInputView',
+    identifyingFields: [[
+        'chatWindowOwnerAsNewMessageFormInputOwner',
+        'discussAsChannelMobileAddItemHeaderOwner',
+        'discussAsChatMobileAddItemHeaderOwner',
+        'discussSidebarCategoryAsAddingItemInputViewOwner',
+    ]],
+    lifecycleHooks: {
+        _willDelete() {
+            if (this.discussAsChannelMobileAddItemHeaderOwner) {
+                this.discussAsChannelMobileAddItemHeaderOwner.clearIsAddingItem();
+            }
+        },
+    },
+    recordMethods: {
+        /**
+         * @param {MouseEvent} ev
+         */
+        onBlur(ev) {
+            if (!this.exists()) {
+                return;
+            }
+            if (this.chatWindowOwnerAsNewMessageFormInputOwner) {
+                return;
+            }
+            this.delete();
+        },
+        onComponentUpdate() {
+            if (this.doFocus) {
+                this.component.root.el.focus();
+                this.update({ doFocus: false });
+            }
+        },
+        onFocusin() {
+            if (!this.exists()) {
+                return;
+            }
+            this.update({ isFocused: true });
+        },
+        /**
+         * @param {MouseEvent} ev
+         */
+        onKeydown(ev) {
+            if (ev.key === 'Escape') {
+                this.delete();
+            }
+        },
```

Comment:
Also not correct for chat window, unless there was a bi-directional `isCausal` (which we probably want to avoid), but otherwise the event should buble up to chat window and let it handle its own destroy.

Reply:
Preemptive return in the case chat window.